pkg/executor/aggfuncs/spill_serialize_helper.go

```diff
@@ -0,0 +1,255 @@
+	buf []byte
+}
+
+// newSpillSerializeHelper creates a new SpillSerializeHelper
+func newSpillSerializeHelper() *spillSerializeHelper {
+	return &spillSerializeHelper{
+		buf: make([]byte, 1024),
+	}
+}
+
+func (s *spillSerializeHelper) serializePartialResult4Count(value partialResult4Count) []byte {
+	spill.SerializeInt64(value, s.buf[0:])
+	return s.buf[:int64Len]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinInt(value partialResult4MaxMinInt) []byte {
+	spill.SerializeInt64(value.val, s.buf[0:int64Len])
+	end := int64Len + boolLen
+	spill.SerializeBool(value.isNull, s.buf[int64Len:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinUint(value partialResult4MaxMinUint) []byte {
+	spill.SerializeUint64(value.val, s.buf[0:uint64Len])
+	end := uint64Len + boolLen
+	spill.SerializeBool(value.isNull, s.buf[uint64Len:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinDecimal(value partialResult4MaxMinDecimal) []byte {
+	*(*types.MyDecimal)(unsafe.Pointer(&s.buf[0])) = value.val
+	end := types.MyDecimalStructSize + boolLen
+	spill.SerializeBool(value.isNull, s.buf[types.MyDecimalStructSize:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinFloat32(value partialResult4MaxMinFloat32) []byte {
+	spill.SerializeFloat32(value.val, s.buf[0:float32Len])
+	end := float32Len + boolLen
+	spill.SerializeBool(value.isNull, s.buf[float32Len:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinFloat64(value partialResult4MaxMinFloat64) []byte {
+	spill.SerializeFloat64(value.val, s.buf[0:float64Len])
+	end := float64Len + boolLen
+	spill.SerializeBool(value.isNull, s.buf[float64Len:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinTime(value partialResult4MaxMinTime) []byte {
+	*(*types.Time)(unsafe.Pointer(&s.buf[0])) = value.val
+	end := timeLen + boolLen
+	spill.SerializeBool(value.isNull, s.buf[timeLen:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinDuration(value partialResult4MaxMinDuration) []byte {
+	spill.SerializeInt64(int64(value.val.Duration), s.buf[0:int64Len])
+	spill.SerializeInt(value.val.Fsp, s.buf[int64Len:int64Len+intLen])
+	end := int64Len + intLen + boolLen
+	spill.SerializeBool(value.isNull, s.buf[int64Len+intLen:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinString(value partialResult4MaxMinString) []byte {
+	spill.SerializeBool(value.isNull, s.buf[0:boolLen])
+	s.buf = s.buf[:boolLen]
+	s.buf = append(s.buf, value.val...)
+	return s.buf
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinJSON(value partialResult4MaxMinJSON) []byte {
+	spill.SerializeBool(value.isNull, s.buf[0:])
+
+	// Assign the return value to `s.buf` is necessary, so that the `s.buf` could expand it's capacity.
```

Comment:
> maybe you can pass `spillSerializeHelper` to `SerializeXXX` function, so the `SerializeXXX` function can update `s.buf` inside, and all the `SerializeXXX` function can have the same signature

- `spillSerializeHelper` in this pr is specifically used for agg function, I think we should not pass it to a common utility.
- Passing `spillSerializeHelper` will make the util package refers to agg package, it will lead to circle reference.

I think pass buffer's pointer is better. And it's reasonable to pass pointer to serialize a variable length data type.